contest/templatetags/contest_extras.py

```diff
@@ -32,3 +38,72 @@ def has_auth(user,contest_id):
     return user_info.has_c_ownership(user,contest)
 
 register.filter("has_auth",has_auth)
+
+#check if user is judge or admin
+@register.filter
+def has_judge_auth(user):
+    if user.is_authenticated():
+        return user.has_judge_auth()
+    else:
+        return False
+register.filter("has_judge_auth",has_judge_auth)
+
+#scoreboard
+@register.filter
+def users_sorted_by_penalty(scoreboard):
+    scoreboard.sort_users_by_penalty()
+    return scoreboard.users
+register.filter("users_sorted_by_penalty",users_sorted_by_penalty)
+
+@register.filter
+def users_sorted_by_solved_testcases(scoreboard):
+    scoreboard.sort_users_by_solved_testcases()
+    return scoreboard.users
+register.filter("users_sorted_by_solved_testcases",users_sorted_by_solved_testcases)
+
+def total_contestant(scoreboard):
+    return scoreboard.users.__len__()
+register.filter("total_contestant",total_contestant)
+
+@register.filter
+def get_problem(scoreboard,id):
+    return scoreboard.get_problem(id)
+
+register.filter("get_problem",get_problem)
+
+@register.filter
+def total_testcase(scoreboard_problem):
+    return scoreboard_problem.total_testcase
+
+register.filter("total_testcase",total_testcase)
+
+@register.filter
+def is_solved(problem):
+    if problem.is_solved():
+        return 1
+    return 0
+register.filter("is_solved",is_solved)
+
+@register.filter
+def testcases_solved(problem):
+    return problem.testcases_solved()
+
+register.filter("testcases_solved",testcases_solved)
+
+@register.filter
+def submit_times(problem):
+    return problem.submit_times()
+
+register.filter("submit_times",submit_times)
+
+@register.filter
+def problem_solved(user):
+    return user.solved()
+
+register.filter("problem_solved",problem_solved)
+
+@register.filter
+def penalty(user,start_time):
+    return user.penalty(start_time)
+
+register.filter("penalty",penalty)
```

Comment:
Both will work.
I think it can be thought as a set, if arranged like this.
If any of them is not registered, it is easier to notice.
Above are my opinion.
If put all of them is the better arrangement, then I'll do that after final decision.